src/codecs/tga/encoder.rs

```diff
@@ -71,22 +176,48 @@ impl<W: Write> TgaEncoder<W> {
             .map_err(|_| ImageError::from(EncoderError::HeightInvalid(height)))?;
 
         // Write out TGA header.
-        let header = Header::from_pixel_info(color_type, width, height)?;
+        let header = Header::from_pixel_info(color_type, width, height, self.use_rle)?;
         header.write_to(&mut self.writer)?;
 
-        // Write out Bgr(a)8 or L(a)8 image data.
-        match color_type {
-            ColorType::Rgb8 | ColorType::Rgba8 => {
-                let mut image = Vec::from(buf);
+        let image_type = ImageType::new(header.image_type);
 
-                for chunk in image.chunks_mut(usize::from(color_type.bytes_per_pixel())) {
-                    chunk.swap(0, 2);
-                }
+        match image_type {
+            //TODO: support RunColorMap, and change match to image_type.is_encoded()
```

Comment:
nit: is_encoded got renamed

Reply:
Has it? Where exactly, because I didn't change it

Reply:
I think it was in an earlier version of the PR. Don't worry about it